### PR TITLE
Issue#24 - Support for bool filter

### PR DESCRIPTION
**Pull request by pshail**
Added boolfitler and updated tools.go
Added testcases
Fixed formatting issues in tools/reflect.go

Aims at fixing #24

### CODE REVIEW

File: yaormfilter/boolfilter_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package yaormfilter_test
+
+import (
+	"github.com/geoffreybauduin/yaorm/yaormfilter"
```
Comment (geoffreybauduin) [inline yaormfilter/boolfilter_test.go]: can you separate golang packages from community packages with a blank line?

File: yaormfilter/boolfilter.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package yaormfilter
+
+import (
+	"github.com/geoffreybauduin/yaorm/tools"
```
Comment (geoffreybauduin) [inline yaormfilter/boolfilter.go]: can you separate golang packages from community packages with a blank line?

File: yaormfilter/boolfilter_test.go (hunk 1 above)
Comment (heckdevice) [reply yaormfilter/boolfilter_test.go]: 👍

File: yaormfilter/boolfilter.go (hunk 2 above)
Comment (heckdevice) [reply yaormfilter/boolfilter.go]: Doing now...

**Comment by heckdevice** [comment]
@geoffreybauduin Pushed the import grouping fix

**Pull request merged**